sql/moz-fx-data-shared-prod/monitoring_derived/bigquery_tables_inventory_v1/query.py

```diff
@@ -50,13 +98,18 @@ def main():
     client.delete_table(destination_table, not_found_ok=True)
 
     for project in args.source_projects:
+
+        create_last_modified_tmp_table(project, tmp_table_name)
+
         client = bigquery.Client(project)
-        query = create_query(project)
+        query = create_query(project, tmp_table_name)
         job_config = bigquery.QueryJobConfig(
             destination=destination_table, write_disposition="WRITE_APPEND"
         )
         client.query(query, job_config=job_config).result()
 
+    client.delete_table(tmp_table_name, not_found_ok=True)
```

Comment:
Maybe we should wrap this inside a `try & finally` block. Just to make sure the delete always takes place. Otherwise, we seem to already handle the case when the table already exists. Is there a need to do this twice?

Reply:
This is to delete the temp table at the end after the destination table has been written.